lang/cs/Org.Apache.REEF.IMRU/OnREEF/Client/REEFIMRUClient.cs

```diff
@@ -0,0 +1,123 @@
+            })
+                .BindNamedParameter(typeof(SerializedMapConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.MapFunctionConfiguration))
+                .BindNamedParameter(typeof(SerializedUpdateConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.UpdateFunctionConfiguration))
+                .BindNamedParameter(typeof(SerializedMapInputCodecConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.MapInputCodecConfiguration))
+                .BindNamedParameter(typeof(SerializedMapInputPipelineDataConverterConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.MapInputPipelineDataConverterConfiguration))
+                .BindNamedParameter(typeof(SerializedUpdateFunctionCodecsConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.UpdateFunctionCodecsConfiguration))
+                .BindNamedParameter(typeof(SerializedMapOutputPipelineDataConverterConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.MapOutputPipelineDataConverterConfiguration))
+                .BindNamedParameter(typeof(SerializedReduceConfiguration),
+                    _configurationSerializer.ToString(jobDefinition.ReduceFunctionConfiguration))
+                .Build();
+
+            IConfiguration groupCommDriverConfig = TangFactory.GetTang().NewConfigurationBuilder()
+                .BindStringNamedParam<GroupCommConfigurationOptions.DriverId>(driverId)
+                .BindStringNamedParam<GroupCommConfigurationOptions.MasterTaskId>(IMRUConstants.UpdateTaskName)
+                .BindStringNamedParam<GroupCommConfigurationOptions.GroupName>(IMRUConstants.CommunicationGroupName)
+                .BindIntNamedParam<GroupCommConfigurationOptions.FanOut>(
+                    IMRUConstants.TreeFanout.ToString(CultureInfo.InvariantCulture)
+                        .ToString(CultureInfo.InvariantCulture))
+                .BindIntNamedParam<GroupCommConfigurationOptions.NumberOfTasks>(
+                    (jobDefinition.NumberOfMappers + 1).ToString(CultureInfo.InvariantCulture))
+                .BindImplementation(GenericType<IGroupCommDriver>.Class, GenericType<GroupCommDriver>.Class)
+                .Build();
+
+            imruDriverConfiguration = Configurations.Merge(imruDriverConfiguration, groupCommDriverConfig);
+
+            // The JobSubmission contains the Driver configuration as well as the files needed on the Driver.
+            var imruJobSubmission = _jobSubmissionBuilderFactory.GetJobSubmissionBuilder()
+                .AddDriverConfiguration(imruDriverConfiguration)
+                .AddGlobalAssemblyForType(typeof(IMRUDriver<TMapInput, TMapOutput, TResult>))
+                .SetJobIdentifier(jobDefinition.JobName)
+                .Build();
+
+            _reefClient.Submit(imruJobSubmission);
+
+            return null;
```

Comment:
no..for onREEF it should actually return either null or reference to the local or hdfs file where results are stored.